gssapi/raw/ext_rfc4178.pyx

```diff
@@ -0,0 +1,54 @@
+GSSAPI="BASE"  # This ensures that a full module is generated by Cython
+
+from gssapi.raw.cython_types cimport *
+from gssapi.raw.cython_converters cimport c_get_mech_oid_set
+from gssapi.raw.creds cimport Creds
+
+from gssapi.raw.misc import GSSError
+
+cdef extern from "python_gssapi_ext.h":
+    OM_uint32 gss_set_neg_mechs(
+        OM_uint32 *minor_status,
+        gss_cred_id_t cred_handle,
+        const gss_OID_set mechs) nogil
+
+
+def set_neg_mechs(Creds cred_handle not None, mechs not None):
+    """
+    set_neg_mechs(Creds cred_handle not None, mechs=None)
+
+    This allows callers to specify the set of security mechanisms that
+    may be negotiated with the credential identified by cred_handle.
+    This call is intended to support specialized callers who need to
+    restrict the set of negotiable security mechanisms from the set of
+    all security mechanisms available to the caller (based on available
+    credentials). Note that if more than one mechanism is specified in
+    mech_set, the order in which those mechanisms are specified implies a
+    relative preference.
+
+    Args:
+        cred_handle (Creds): the Credentials to query
```

Comment:
Technically not a query - that would be more true of `gss_get_neg_mechs()` (which no one implements).